File: frontend/src/types/task.ts
export type TaskStatus = "pending" | "in_progress" | "completed";

export interface Project {
  id: number;
  name: string;
  description?: string;
  owner_id: number;
}

export interface Task {
  id: number;
  title: string;
  description?: string;
  status: TaskStatus;
  project_id: number;
  responsible_user_id: number;
}

export interface TaskCreate {
  title: string;
  description?: string;
}

export interface TaskStatusUpdate {
  status: TaskStatus;
}

File: backend/app/schemas/task.py
from typing import Literal, Optional
from pydantic import BaseModel

TaskStatusType = Literal["pending", "in_progress", "completed"]


class TaskBase(BaseModel):
    title: str
    description: Optional[str] = None


class TaskCreate(TaskBase):
    pass


class TaskUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    status: Optional[TaskStatusType] = None


class TaskStatusUpdate(BaseModel):
    status: TaskStatusType


class Task(TaskBase):
    id: int
    status: TaskStatusType
    project_id: int
    responsible_user_id: int

    model_config = {"from_attributes": True}
